### PR TITLE
feat(Gallery): Convert gallery to typescript

### LINKED ISSUE #2431: Convert Gallery to Typescript
Convert gallery to typescript, add demo, and integration testing

### PR DESCRIPTION
**Pull request by jessiehuff**
Fixes #2431

### CODE REVIEW

File: packages/patternfly-4/react-core/src/layouts/Gallery/Gallery.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/layouts/Gallery/gallery';
+
+export interface GalleryProps extends React.HTMLProps<HTMLDivElement> {
+  /** content rendered inside the Gallery layout */
+  children?: React.ReactNode; 
+  /** additional classes added to the Gallery layout */
+  className?: string; 
+  /** Adds space between children. */
+  gutter?: boolean;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/layouts/Gallery/Gallery.tsx]: This does not make sense form previous implementation of gallery.  Shouldn't it be a union? `sm|md|lg`?

why is gutter now a boolean.  This is a breaking change.

File: packages/patternfly-4/react-core/src/layouts/Gallery/Gallery.test.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import React from 'react';
-import Gallery from './Gallery';
+import * as React from 'react';
+import { Gallery } from './Gallery';
 import { GutterSize } from '../../styles/gutters';
 import { shallow } from 'enzyme';
 
 Object.values(GutterSize).forEach(gutter => {
   test(`gutter ${gutter}`, () => {
-    const view = shallow(<Gallery gutter={gutter} />);
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/layouts/Gallery/Gallery.test.tsx]: This does not make sense to me.  You cant have `const view = shallow(<Gallery sm/>);` or `const view = shallow(<Gallery lg/>);` etc

File: packages/patternfly-4/react-core/src/layouts/Gallery/Gallery.tsx (hunk 1 above)
Comment (jessiehuff) [reply packages/patternfly-4/react-core/src/layouts/Gallery/Gallery.tsx]: Currently sm, md, lg doesn't do anything other than resolve to true. Based on conversation, I thought we weren't going to continue passing the string and make it a boolean, but I can change them all back to sm, md, lg and just change them again later if and when we choose to fix the gutter sizes.

File: packages/patternfly-4/react-core/src/layouts/Gallery/examples/Gallery.md [hunk 3]
```diff
@@ -30,7 +31,7 @@ import { Gallery, GalleryItem } from '@patternfly/react-core';
 import React from 'react';
 import { Gallery, GalleryItem } from '@patternfly/react-core';
 
-<Gallery gutter="md">
+<Gallery gutter>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/layouts/Gallery/examples/Gallery.md]: does this cause error in demo.  I think it should be reverted

**Pull request merged**